### PR TITLE
fix: Support minimum of node v14

### PR DESCRIPTION
**Pull request by schwern**
* Increase node to >= v14.
  * v10 and v12 are EOL and Jest dropped support for them.
* Matrix test node v14, v16, and v18
  * Only do the README and commit lint once.
* Fix the examples for v14.

### CODE REVIEW

File: examples/index.ts [hunk 1]
```diff
@@ -92,6 +92,7 @@ main()
   .then(() => {
     console.log('success!!');
   })
-  .catch(e => {
-    console.error('failed to get from cache', e);
+  .catch((e: Error) => {
+    console.error(`failed to get from cache ${e.message}`);
+    throw e;
```
Comment (cprice404) [inline examples/index.ts:97]: thank you for fixing this!  I have noticed that we are missing the throws for these in some places and it makes it really confusing to debug :)

File: test/cache-get.test.ts [hunk 2]
```diff
@@ -1,19 +1,21 @@
...
+import {TextEncoder} from 'util';
+
+const TEXT_ENCODER = new TextEncoder();
 
 describe('CacheGet', () => {
   describe('#toString()', () => {
     it('shows a short value', () => {
       const value = 'Napoleon';
-      const hit = new CacheGet.Hit(Buffer.from(value));
+      const hit = new CacheGet.Hit(TEXT_ENCODER.encode(value));
```
Comment (cprice404) [inline test/cache-get.test.ts:10]: Why the switch from buffer to text encoder?

**Review by cprice404 — COMMENTED** [review]
thanks for doing this!  It's a really important change.  I just had one question about buffer vs. textencoder.

File: examples/index.ts (hunk 1 above)
Comment (schwern) [reply examples/index.ts:97]: I should make them consistent and all throw?

File: test/cache-get.test.ts (hunk 2 above)
Comment (schwern) [reply test/cache-get.test.ts:10]: Node 10 and 12 don't have it. Now that we're dropping Node 10 and 12 maybe we should switch from TextEncoder to Buffer. However, TextEncoder is what we use everywhere else, and I don't know the benefits/drawbacks of Buffer vs TextEncoder, so switching is a bigger change.

File: examples/index.ts (hunk 1 above)
Comment (cprice404) [reply examples/index.ts:97]: I was talking about in other repos where we use TS for infra, I'm not aware of whether there are any more missing in this repo.  If you audited and they all look good in this repo, great!

File: test/cache-get.test.ts (hunk 2 above)
Comment (cprice404) [reply test/cache-get.test.ts:10]: okay

**Review by cprice404 — APPROVED** [review]

**Pull request merged**